修改core.py中upload_chunk函数以在上传发生异常后重试

Asa-Chiri opened the pull request:
将整个upload_chunk函数中的上传部分用try包裹并捕获异常返回False，使得程序可以在上传发生异常后重试，而不是像现在一样直接退出。

Asa-Chiri committed:
Update core.py

将整个upload_chunk函数中的上传部分用try包裹并捕获异常返回False，使得程序可以在上传发生异常后重试.

FortuneDayssss commented:
谢谢提交~

upload_chunk在被调用的时候一般是被Retry装饰器包装的
我认为是不是把Try Except加到Retry装饰器里可读性会比较好一些？
可以麻烦你把Try Except改到util/retry.py里面吗，然后我merge，谢谢

Commit:
Revert "Update core.py"

This reverts commit 7e604cbfadea362cfdc67a8546de31df956df8e1.

Commit:
将对于core.py中关于Exception的修改移到utils/retry.py中

Revert了对于core.py的修改；
修改utils/retry.py中的Retry装饰器run函数，在return_value = func(*args, **kwargs)发生exception时捕获并令return_value为not self.success_return_value使发生exception时可以重试。

Asa-Chiri commented:
> 谢谢提交~
> 
> upload_chunk在被调用的时候一般是被Retry装饰器包装的
> 我认为是不是把Try Except加到Retry装饰器里可读性会比较好一些？
> 可以麻烦你把Try Except改到util/retry.py里面吗，然后我merge，谢谢

感谢建议，我已经将相关逻辑移动到了utils/retry.py中，烦请Review~详情见commit信息
